morphman/automated_landmarking/automated_landmarking_bogunovic.py

```diff
@@ -0,0 +1,175 @@
+    k2 = get_point_data_array("k2", line)
+
+    # Remove a min / max point that is in reality a saddle point
+    for i in min_point_ids:
+        for j in max_point_ids:
+            if abs(i - j) < 5 and abs(curvature[i] - curvature[j]) < 0.01:
+                if i in min_point_ids:
+                    min_point_ids.remove(i)
+                if j in max_point_ids:
+                    max_point_ids.remove(j)
+
+    k1_points = k1[max_point_ids]
+    k2_points = k2[max_point_ids]
+    k_points = np.zeros((k1_points.shape[0], 2))
+    k_points[:, 0] = k1_points[:, 0]
+    k_points[:, 1] = k2_points[:, 0]
+    theta = np.zeros(k1_points.shape[0] - 1)
+    for i in range(theta.shape[0]):
+        a = k_points[i, :] / np.sqrt(np.sum(k_points[i, :] ** 2))
+        b = k_points[i + 1, :] / np.sqrt(np.sum(k_points[i + 1, :] ** 2))
+        theta[i] = math.acos(np.dot(a, b))
+        theta[i] = theta[i] * 180 / math.pi
+
+    # Tolerance parameters from Bogunovic et al. (2012)
+    tol_ant_post = 60
+    tol_sup_ant = 45
+    tol_post_inf = 45
+    tol_inf_end = 110
+
+    # Find coronal coordinate and maximum (within anterior bend)
+    coordinates = get_centerline_coordinates(line, length)
+    coronal_coordinate = coordinates[coronal_axis]
+    max_coronal_coordinate_id = get_maximum_coronal_coordinate(coronal_coordinate, length)
+
+    def find_interface(start, direction, tol, part):
+        stop = direction if direction == -1 else theta.shape[0]
+        success = False
+        for i in range(start - 1, stop, direction):
+            if theta[i] > tol:
+                success = True
+                break
+
+        if success:
+            start = max_point_ids[i]
+            stop = max_point_ids[i + 1]
+            index = ((min_point_ids > start) * (min_point_ids < stop)).nonzero()[0].max()
+            min_point = (min_point_ids[index])
+            interfaces[part] = min_point
+
+        elif not success and part == "sup_ant":
+            print("-- Where not able to identify the interface between the " +
+                  "anterior and superior bend. Check the coronal coordinates")
+            return None
+
+        elif not success and part != "inf_end":
+            print("-- The geometry is to short to be classified with superior" +
+                  ", anterior, posterior and inferior.")
+            return None
+
+        elif not success and part == "inf_end":
+            interfaces["inf_end"] = np.array([0])
+            i = 0
+            print("-- End of inferior is at the end of the geometry, this might" +
+                  "affect the geometry stats")
+        else:
+            print("-- Something happened, idea: some bend ended at the last point")
+            return None
+
+        return i
+
+    interfaces = {}
+    min_point_ids = np.array(min_point_ids)
+
+    index = np.array((max_coronal_coordinate_id > max_point_ids).nonzero()[0]).max()
+    start = find_interface(index, -1, tol_ant_post, "ant_post")
```

Comment:
I'm wondering if "anterior_posterior" would be a better variable name? The same goes for the other landmarks.

Reply:
Yes, this is more explicit. Will change for all landmarks.